File: c/cz/mg/vulkan/PFN_vkGetImageSubresourceLayout.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkGetImageSubresourceLayout_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong image, jlong pSubresource, jlong pLayout){
    (void)env;
    (void)jc;
    PFN_vkGetImageSubresourceLayout f = (PFN_vkGetImageSubresourceLayout)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkImage*)jniLongToPointer(image)),
        ((VkImageSubresource*)jniLongToPointer(pSubresource)),
        ((VkSubresourceLayout*)jniLongToPointer(pLayout))
    );
}

File: cz/mg/vulkan/PFNvkGetImageSubresourceLayout.java
package cz.mg.vulkan;

public class PFNvkGetImageSubresourceLayout extends VkFunctionPointer {
    public PFNvkGetImageSubresourceLayout() {
    }

    public PFNvkGetImageSubresourceLayout(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetImageSubresourceLayout(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkGetImageSubresourceLayout(long value) {
        setValue(value);
    }

    public PFNvkGetImageSubresourceLayout(VkInstance instance) {
        super(instance, new VkString("vkGetImageSubresourceLayout"));
    }

    public void call(VkDevice device, VkImage image, VkImageSubresource pSubresource, VkSubresourceLayout pLayout){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getSinkAddress(), image != null ? image.getVkAddress() : VkPointer.getSinkAddress(), pSubresource != null ? pSubresource.getVkAddress() : VkPointer.NULL, pLayout != null ? pLayout.getVkAddress() : VkPointer.NULL);
    }

    private static native void call(long vkaddress, long device, long image, long pSubresource, long pLayout);
}
